tests/unit/server/api/v1/test_suggestions.py

```diff
@@ -0,0 +1,76 @@
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import TYPE_CHECKING
+from uuid import uuid4
+
+import pytest
+from argilla._constants import API_KEY_HEADER_NAME
+from argilla.server.models import Suggestion, UserRole
+from sqlalchemy import func, select
+
+from tests.factories import SuggestionFactory, UserFactory
+
+if TYPE_CHECKING:
+    from httpx import AsyncClient
+    from sqlalchemy.ext.asyncio import AsyncSession
+
+
+@pytest.mark.asyncio
+class TestSuiteSuggestions:
+    @pytest.mark.parametrize("role", [UserRole.admin, UserRole.owner])
+    async def test_delete_suggestion(self, async_client: "AsyncClient", db: "AsyncSession", role: UserRole) -> None:
+        suggestion = await SuggestionFactory.create()
+        user = await UserFactory.create(role=role, workspaces=[suggestion.record.dataset.workspace])
+
+        response = await async_client.delete(
+            f"/api/v1/suggestions/{suggestion.id}",
+            headers={API_KEY_HEADER_NAME: user.api_key},
+        )
+
+        assert response.status_code == 200
+        assert response.json() == {
+            "id": f"{suggestion.id}",
+            "question_id": f"{suggestion.question_id}",
+            "type": None,
+            "score": None,
+            "value": "negative",
+            "agent": None,
+        }
+        assert (await db.execute(select(func.count(Suggestion.id)))).scalar() == 0
+
+    async def test_delete_suggestion_non_existent(self, async_client: "AsyncClient", owner_auth_header: dict) -> None:
+        response = await async_client.delete(f"/api/v1/suggestions/{uuid4()}", headers=owner_auth_header)
+
+        assert response.status_code == 404
+
+    async def test_delete_suggestion_as_admin_from_another_workspace(self, async_client: "AsyncClient") -> None:
+        suggestion = await SuggestionFactory.create()
+        user = await UserFactory.create(role=UserRole.admin)
+
+        response = await async_client.delete(
+            f"/api/v1/suggestions/{suggestion.id}", headers={API_KEY_HEADER_NAME: user.api_key}
+        )
+
+        assert response.status_code == 403
+
+    async def test_delete_suggestion_as_annotator(self, async_client: "AsyncClient") -> None:
+        suggestion = await SuggestionFactory.create()
+        user = await UserFactory.create(role=UserRole.annotator, workspaces=[suggestion.record.dataset.workspace])
+
+        response = await async_client.delete(
+            f"/api/v1/suggestions/{suggestion.id}", headers={API_KEY_HEADER_NAME: user.api_key}
+        )
+
+        assert response.status_code == 403
```

Comment:
Maybe for an upcoming iteration, but we can parametrise the conditions that trigger HTTP 403 as in most of the cases i.e. `annotator` or `admin` from another workspace